src/lib.rs

```diff
@@ -446,11 +446,26 @@ impl<Types: IpfsTypes> Ipfs<Types> {
     }
 
     /// Resolves a ipns path to an ipld path.
-    pub async fn resolve_ipns(&self, path: &IpfsPath) -> Result<IpfsPath, Error> {
-        self.ipns()
-            .resolve(path)
-            .instrument(self.span.clone())
-            .await
+    pub async fn resolve_ipns(&self, path: &IpfsPath, recursive: bool) -> Result<IpfsPath, Error> {
+        let ipns = self.ipns();
+        let mut resolved = ipns.resolve(path).await;
+
+        if recursive {
+            let mut previous = None;
+            while let Ok(ref res) = resolved {
+                if let Some(ref prev) = previous {
+                    if prev == res {
+                        break;
+                    }
+                }
+                previous = Some(res.clone());
+                resolved = ipns.resolve(&res).await;
```

Comment:
I think here should be a hashmap of seen results. I can at least imagine it can be easy to create a loop with two domains.

Reply:
good point :+1: